m4/curl-compilers.m4

```diff
@@ -903,7 +903,14 @@ AC_DEFUN([CURL_SET_COMPILER_WARNING_OPTS], [
           #
           dnl Only clang 3.2 or later
           if test "$compiler_num" -ge "302"; then
-            tmp_CFLAGS="$tmp_CFLAGS -Wmissing-variable-declarations"
+            case $host_os in
+            cygwin* | mingw*)
+              dnl libtool wrapper executable throws these warnings
```

Comment:
this is fine except I suggest be more specific about what "these warnings" for posterity, for example
              dnl skip missing-variable-declarations warnings for cygwin
              dnl and mingw because libtool causes them